app/services/school_data_importer.rb

```diff
@@ -0,0 +1,70 @@
+require "open-uri"
+require "csv"
+
+class SchoolDataImporter
+  def run
+    download_and_save_csv
+
+    CSV.foreach(csv_file_name, headers: true, encoding: "windows-1251:utf-8") do |row|
+      local_authority = LocalAuthority.find_or_create_by(
+        code: row["LA (code)"],
+        name: row["LA (name)"]
+      )
+
+      school = row_to_school(row)
+
+      school.local_authority = local_authority
+
+      unless school.valid?
+        Rails.logger.warn("Failed to import school: #{school.urn} - #{school.errors.full_messages}") && next
+      end
+      school.save
+    end
+
+    delete_csv_file
+  rescue OpenURI::HTTPError => e
```

Comment:
Let's not catch these for now. If the download is failing, we want a noisy failure. and if our intention is to make this a background job then failing noisily will mean that the retry mechanism can kick in and intermittent failures are handled automatically.